processors/filterprocessor/filter_processor_test.go

```diff
@@ -0,0 +1,101 @@
+package filterprocessor
+
+import (
+	"context"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/indexer/conduit"
+	"github.com/algorand/indexer/data"
+	"github.com/algorand/indexer/plugins"
+	"github.com/algorand/indexer/processors"
+)
+
+func TestCheckTagExistsAndHasCorrectFunction(t *testing.T) {
+	// check that something that doesn't exist throws an error
+	err := checkTagExistsAndHasCorrectFunction("const", "SignedTxnWithAD.SignedTxn.Txn.PaymentTxnFields.LoreumIpsum.SDF")
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "does not exist in transactions")
+
+	err = checkTagExistsAndHasCorrectFunction("const", "LoreumIpsum")
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "does not exist in transactions")
+
+	// Fee does not have a "String" Function so we cant use const with it.
+	err = checkTagExistsAndHasCorrectFunction("const", "SignedTxnWithAD.SignedTxn.Txn.Header.Fee")
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "does not contain the needed method")
+}
+
+// TestFilterProcessor_Init tests initialization of the filter processor
+func TestFilterProcessor_Init(t *testing.T) {
```

Comment:
I think we need more tests here. I don't see `all` being tested, there's not error tests for example supplying a bad tag, or a field which doesn't have the correct operations defined such as a MicroAlgos field.

Reply:
I added more unit tests.  The test that detects either a bad tag or a field that doesnt have the correct function is located in searcher_test.go